core/test/reorder/amd.cpp

```diff
@@ -0,0 +1,276 @@
+{
+    using matrix_type = typename TestFixture::matrix_type;
+    using index_type = typename TestFixture::index_type;
+    this->forall_matrices([this] {
+        this->forall_modes([this] {
+            auto perm = this->amd->generate(this->mtx);
+
+            auto perm_array = gko::make_array_view(this->ref, this->num_rows,
+                                                   perm->get_permutation());
+            GKO_ASSERT_ARRAY_EQ(perm_array, this->permutation_ref);
+
+            auto permuted_mtx =
+                gko::as<matrix_type>(this->mtx->permute(&perm_array));
+            std::unique_ptr<gko::factorization::elimination_forest<index_type>>
+                forest;
+            std::unique_ptr<matrix_type> factorized_mtx;
+            std::unique_ptr<matrix_type> factorized_permuted_mtx;
+            gko::factorization::symbolic_cholesky(this->mtx.get(), true,
+                                                  factorized_mtx, forest);
+            gko::factorization::symbolic_cholesky(
+                permuted_mtx.get(), true, factorized_permuted_mtx, forest);
+            int fillin_mtx = factorized_mtx->get_num_stored_elements() -
+                             this->mtx->get_num_stored_elements();
+            int fillin_permuted =
+                factorized_permuted_mtx->get_num_stored_elements() -
+                permuted_mtx->get_num_stored_elements();
+            ASSERT_LE(fillin_permuted, fillin_mtx - this->fillin_reduction);
+        });
+    });
+}
+
+
+TYPED_TEST(Amd, ReducesFillInAni4)
+{
+    using matrix_type = typename TestFixture::matrix_type;
+    using index_type = typename TestFixture::index_type;
+    this->mtx = gko::read<matrix_type>(
+        std::ifstream{gko::matrices::location_ani4_mtx}, this->ref);
+    this->num_rows = this->mtx->get_size()[0];
+    this->amd =
+        gko::experimental::reorder::Amd<index_type>::build().on(this->ref);
+
+    auto perm = this->amd->generate(this->mtx);
+
+    auto perm_array = gko::make_array_view(this->ref, this->num_rows,
+                                           perm->get_permutation());
+    auto permuted_mtx = gko::as<matrix_type>(this->mtx->permute(&perm_array));
+    std::unique_ptr<gko::factorization::elimination_forest<index_type>> forest;
+    std::unique_ptr<matrix_type> factorized_mtx;
+    std::unique_ptr<matrix_type> factorized_permuted_mtx;
+    gko::factorization::symbolic_cholesky(this->mtx.get(), true, factorized_mtx,
+                                          forest);
+    gko::factorization::symbolic_cholesky(permuted_mtx.get(), true,
+                                          factorized_permuted_mtx, forest);
+    int fillin_mtx = factorized_mtx->get_num_stored_elements() -
+                     this->mtx->get_num_stored_elements();
+    int fillin_permuted = factorized_permuted_mtx->get_num_stored_elements() -
+                          permuted_mtx->get_num_stored_elements();
+    ASSERT_LE(fillin_permuted, fillin_mtx * 2 / 5);
+}
```

Comment:
```suggestion
}


}  // namespace
```

Reply:
I'm not using an unnamed namespace around the tests here